Tests/Microsoft.AppCenter.Crashes.Test.Windows/HandledErrorTest.cs

```diff
@@ -61,6 +81,124 @@ public void TrackErrorWithValidParameters()
+                .Callback((string uri, string method, IDictionary<string, string> headers, string content, CancellationToken cancellation) =>
+                {
+                    count++;
+                    if (count == ExpectedLogCount)
+                    {
+                        semaphore.Release();
+                    }
+                }).ReturnsAsync("");
+
+            var exception = new System.Exception("Something went wrong.");
+            var properties = new Dictionary<string, string> { { "k1", "v1" }, { "p2", "v2" } };
+            var attachmentLog = GetValidErrorAttachmentLog();
+            var attachmentLog2 = GetValidErrorAttachmentLog();
+            Crashes.TrackError(exception, properties, attachmentLog, attachmentLog2);
+
+            // Wait until the http layer sends the log.
+            semaphore.Wait(2000);
+            Assert.AreEqual(ExpectedLogCount, count);
+        }
+
+        [TestMethod]
+        public void TrackErrorWithoutPropertiesWithAttachments()
+        {
+            var semaphore = new SemaphoreSlim(0);
+            HandledErrorLog actualLog = null;
+            ErrorAttachmentLog actualErrorAttachmentLog = null;
+            Mock.Get(_mockNetworkAdapter).Setup(adapter => adapter.SendAsync(
+                It.IsAny<string>(),
+                It.IsAny<string>(),
+                It.IsAny<IDictionary<string, string>>(),
+                It.IsAny<string>(),
+                It.IsAny<CancellationToken>()))
+                .Callback((string uri, string method, IDictionary<string, string> headers, string content, CancellationToken cancellation) =>
+                {
+                    var log = JsonConvert.DeserializeObject<LogContainer>(content, LogSerializer.SerializationSettings).Logs[0];
+                    if (log.GetType() == typeof(ErrorAttachmentLog))
+                    {
+                        actualErrorAttachmentLog = log as ErrorAttachmentLog;
+                    }
+                    else
+                    {
+                        actualLog = log as HandledErrorLog;
+                    }
+                    if (actualLog != null && actualErrorAttachmentLog != null)
+                    {
+                        semaphore.Release();
+                    }
+                }).ReturnsAsync("");
+
+            var exception = new System.Exception("Something went wrong.");
+            var attachmentLog = GetValidErrorAttachmentLog();
+            Crashes.TrackError(exception, null, attachmentLog);
+
+            // Wait until the http layer sends the log.
+            semaphore.Wait(2000);
+            Assert.IsNotNull(actualLog);
+            Assert.AreEqual(exception.Message, actualLog.Exception.Message);
+            Assert.IsNotNull(actualErrorAttachmentLog);
+            Assert.AreEqual(actualLog.Id, actualErrorAttachmentLog.ErrorId);
+            CollectionAssert.AreEquivalent(attachmentLog.Data, actualErrorAttachmentLog.Data);
+        }
```

Comment:
What is the expected behavior if there are too many properties but valid attachments?

Reply:
Windows SDK will emit a warning log if there are more than 2 attachments but still allow it. See https://github.com/microsoft/appcenter-sdk-dotnet/pull/1170/files#diff-0df124bf3e36f1bf4b13cb6ce6938250R423.

We normally don't test App Center logs in unit tests so we didn't add a test case specifically for that.

Reply:
He's asking about properties. We don't change the behavior based on properties, so the HandledError will send the properties like it does now (chopped off at the max), and the app will continue on to sending the attachments without a problem.